client/src/components/Timer.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { useEffect, useState } from "react";
+import { useGame } from "@/context/GameContext";
+
+export default function Timer() {
+    
+    const { hours, minutes, seconds} = {hours: 0, minutes: 0, seconds: 0};
+    const [[hrs, mins, secs], setTime] = useState([hours, minutes, seconds]);
+    const game = useGame();
+
+    const tick = () => {
+        if (mins === 59 && secs === 59) {
+            setTime([hrs + 1, 0, 0]);
+        } else if (mins !== 59 && secs === 59) {
+            setTime([hrs, mins + 1, 0]);
+        } else {
+            setTime([hrs, mins, secs + 1]);
+        }
+    };
+
+    function reset() {
+        setTime([0, 0, 0]);
+    }
+
+    useEffect(() => {
+        reset();
+    }, [game.level]);
+
+    useEffect(() => {
+        const timerId = setInterval(() => tick(), 1000);
+        return () => clearInterval(timerId);
+    });
```

Comment:
Should add empty dependency array but I think the hook actually depends on the `tick()` method.

Reply:
When I try adding an empty array, the clock stops after 1 second elapses. The problem where the clock stops whenever a change is made to the page might be related to this

Reply:
This was fixed by adding `[secondsElapsed]` to `useEffect`

Reply:
Right now this effect runs conditionally when `setSecondsElapsed` changes which seems incorrect, I tried it out with passing `[]` (indicating that this effect has no dependency on props/state) and it seeeeems to work fine, I would change it back to `[]`.

Reply:
You're right about this, it works as `[]`